### DISCUSSION TITLE
Add `has` filter on relation[]

### DISCUSSION THREAD

**Discussion by codeit-ninja** [prisma/prisma#23944] [Q&A] [answered]
### Question

Why does the where statement on a relational[] query not support `has`, `hasEvery`, `hasSome`, like filtering on `enums` does?
It would make sense if it had the `has` method too.

### How to reproduce (optional)

https://stackoverflow.com/questions/73382548/get-posts-with-selected-tags-using-prisma

### Expected behavior (optional)

It should return posts that have all selected tags + additional tags. Every filters the posts where it just has all tags, and some returns posts that have 1 or more tags.

We are missing the query that combines `.some(row => .include(...))`.

Here is a `JavaScript` array example to visualize the problem

```js
const test = [
    { numbers: [0, 1, 2, 3, 4] },
    { numbers: [4, 5, 6, 2] },
    { numbers: [1, 4, 3, 8] },
    { numbers: [4, 2] },
];

const input = [3, 4];

const output = test.filter(item =>
    input.every(num => item.numbers.includes(num))
);

console.log(output);
// [
//     { numbers: [0, 1, 2, 3, 4] },
//     { numbers: [1, 4, 3, 8] }
// ]
```

**Comment by ludralph** [comment, marked as answer]
Hi @codeit-ninja 👋 

Prisma provides the `some`, `every`, and `none` options to filter records by the properties of related records on the "-to-many" side of the relation. However, the `has`, `hasEvery`, and `hasSome` filters are specifically designed for scalar list fields, not for relational queries.

If you want to see this supported for relational queries, please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your use case so our engineers can take a look.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
